Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py

```diff
@@ -1241,6 +1321,201 @@ def get_processes(client: Client, args: dict) -> CommandResults:
+    tenant = tenant_str.lower() == 'true'
+
+    sort_by = "updatedAt"
+    sort_order = "desc"
+
+    skip = int(args.get('skip', "0"))
+    limit = int(args.get('limit', "100"))
+    group_ids = args.get('group_ids', None)
+    site_ids = args.get('site_ids', None)
+    account_ids = args.get('account_ids', None)
+
+    contents = []
+
+    block_list = client.get_blocklist_request(tenant=tenant, group_ids=group_ids, site_ids=site_ids,
+                                              account_ids=account_ids, skip=skip, limit=limit,
+                                              sort_by=sort_by, sort_order=sort_order)
+    for block in block_list:
+        contents.append({
+            'CreatedAt': block.get('createdAt'),
+            'Description': block.get('description'),
+            'ID': block.get('id'),
+            'OSType': block.get('osType'),
+            'ScopeName': block.get('scopeName'),
+            'ScopePath': block.get('scopePath'),
+            'Source': block.get('source'),
+            'Type': block.get('type'),
+            'UpdatedAt': block.get('updatedAt'),
+            'UserId': block.get('userId'),
+            'Value': block.get('value')
+        })
+
+    return CommandResults(
+        readable_output=tableToMarkdown('SentinelOne Blocklist', contents, removeNull=True),
+        outputs_prefix='SentinelOne.Blocklist',
+        outputs_key_field='Value',
+        outputs=contents,
+        raw_response=block_list)
+
+# File Fetch Commands
+
+
+def fetch_file(client: Client, args: dict) -> str:
+    """
+    Initiate a file fetch request on an agent
+    """
+    agent_id = args.get('agent_id')
+    file_path = args.get('file_path')
+    password = args.get('password')
+
+    client.fetch_file_request(agent_id, file_path, password)
+    return f"Intiated fetch-file action for {file_path} on Agent {agent_id}"
+
+
+def extract_sentinelone_zip_file(zip_file_data: bytes, password: str) -> Tuple[str, bytes]:
+    """
+    Helper funciton for `download_fetched_file`
+    """
+    file_archive = io.BytesIO(zip_file_data)
+    zip_file = zipfile.ZipFile(file_archive)
+
+    # Each .zip file returned by SentinelOne has a manifest.json file. Then it
+    # Re-creates the folder paths inside the zip, and stores the collected file
+    # (e.g. C/path/to/file.txt inside the zip)
+    #
+    # We assume only one file was collected, since that's how our integration commands are
+    # implemented
+
+    file_name = [name for name in zip_file.namelist() if name != "manifest.json"][0]
+    file_data = zip_file.read(file_name, password.encode('utf-8'))
+    return file_name, file_data
+
+
+def download_fetched_file(client: Client, args: dict) -> List[CommandResults]:
+    """
+    Download a file that has been requested by `fetch-file`
+    """
+    agent_id = args.get('agent_id')
+    activity_id = args.get('activity_id')
+    password = args.get('password')
+    assert isinstance(password, str)
```

Comment:
why do you need that? make that argument required in the command and then you won't need that check.